### DISCUSSION TITLE
409 (cliente già esistente) in fase di generazione fattura

### DISCUSSION THREAD

**Discussion by zofe** [fattureincloud/api#169] [Q&A] [answered]
**Oggetto**:
409 (cliente già esistente) su generazione fattura

**Premessa**: 
l'integrazione che attualmente abbiamo (fra i nostri sistemi e FiC) prevede "anche" la generazione Anagrafiche Cliente via API,  questa pero' non è completamente sincrona: ossia  è possibile che siano già stati anagrafati clienti da interfaccia FiC dei quali i nostri sistemi non hanno il relativo identificativo FiC.

**Caso d'uso dell'errore**:
Il nostro sistema, al momento della generazione di una fattura, non avendo già anagrafato direttamente il cliente tenta di crearlo su FiC, in risposta si ha il generico messaggio 409 (cliente già esistente).

Considerazioni:
Il messaggio in risposta è decisamente generico (non è utile per identificare l'ID dell'anagrafica in conflitto).
Non è chiaro su quale dato/i si basa questo vincolo (partita iva? sdi? pec?) 
Sarebbe utile sapere come prevenire l'errore cercando preventivamente una anagrafica nell'eventualità che non sia stata creata dai nostri sistemi. Anche qui occorre sapere su quale campi effettuare la ricerca, visto che attualmente non mi sembra ci sia univocità in inserimento della partita iva da interfaccia FiC).

**Comment by emajo** [comment, marked as answer]
Ciao @zofe per quanto riguarda l'univocità del cliente ti consiglio di leggere [questo paragrafo](https://developers.fattureincloud.it/docs/guides/client-creation#%EF%B8%8F-client-uniqueness), abbiamo anche una discussion in cui abbiamo già affrontato questo argomento che puoi trovare [qui](https://github.com/fattureincloud/api/discussions/66).
Abbiamo comunque in programma di migliorare la gestione di questo tipo di errore ma attualmente non sappiamo ancora quando verrà fatto, il mio consiglio è comunque quello di utilizzare un code univoco generato da voi, attualmente è la soluzione migliore.
In caso volessi controllare preventivamente l'esistenza di un cliente è possibile fare una list clients passando un filtro che agisca sui tre campi vat_number, tax_code e name.

Facci sapere se hai ancora dei dubbi,
grazie,
Emanuele.
